File: solutions/python/bob/1/bob.py
import re

SILENCE = re.compile('^\s*$')
SHOUTING = re.compile('^[^A-Za-z]*[A-Z]+(?:[^a-z]*)$')
ASKING = re.compile('.*\?\s*$')

def response(message):
    if SILENCE.match(message) is not None:
        return 'Fine. Be that way!'
    
    if SHOUTING.match(message) is not None:
        return "Calm down, I know what I'm doing!" if ASKING.match(message) is not None else 'Whoa, chill out!'

    return 'Sure.' if ASKING.match(message) is not None else 'Whatever.'

File: solutions/csharp/bob/1/Bob.cs
using System;
using System.Text.RegularExpressions;


public static class Bob
{
    static Regex _silence = new Regex(@"^\s*$", RegexOptions.Compiled);
    static Regex _asking = new Regex(@"\?\s*$", RegexOptions.Compiled);
    static Regex _shouting = new Regex(@"^[^A-Za-z]*[A-Z]+(?:[^a-z]*)$", RegexOptions.Compiled);

    public static string Response(string statement)
    {
        if (_silence.IsMatch(statement)) {
            return "Fine. Be that way!";
        } else if (_asking.IsMatch(statement)) {
            return _shouting.IsMatch(statement) ? "Calm down, I know what I'm doing!" : "Sure.";
        } else if (_shouting.IsMatch(statement)) {
            return "Whoa, chill out!";
        } else {
            return "Whatever.";
        }
    }
}
